app/src/ui/branches/branches-container.tsx

```diff
@@ -55,20 +55,21 @@ export class BranchesContainer extends React.Component<
     }
   }
 
-  private onItemClick = (item: Branch) => {
-    this.checkoutBranch(item.nameWithoutRemote)
-  }
-
-  private checkoutBranch(branch: string) {
+  private onItemClick = (branch: Branch) => {
     this.props.dispatcher.closeFoldout(FoldoutType.Branch)
 
     const currentBranch = this.props.currentBranch
 
-    if (!currentBranch || currentBranch.name !== branch) {
+    if (!currentBranch || currentBranch.name !== branch.name) {
```

Comment:
Since `currentBranch` does not seem to be a boolean, we should check explicitly for null (assuming that's the intent here).